src/main/java/com/bloxbean/cardano/client/account/Account.java

```diff
@@ -278,13 +315,50 @@ private void validateMnemonic() {
     }
 
     private HdKeyPair getHdKeyPair() {
-        HdKeyPair hdKeyPair = new CIP1852().getKeyPairFromMnemonic(mnemonic, derivationPath);
+        HdKeyPair hdKeyPair;
+        if (mnemonic == null || mnemonic.trim().length() == 0) {
+            hdKeyPair = new CIP1852().getKeyPairFromAccountKey(this.secretKey, derivationPath);
+        }
+        else {
+            hdKeyPair = new CIP1852().getKeyPairFromMnemonic(mnemonic, derivationPath);
+        }
+        return hdKeyPair;
+    }
+
+    public HdKeyPair getExternalKeyPair() {
```

Comment:
Also,  we can change this method to private as it's not used  outside of this class. 
Currently if you print "Account" object, it shows privatekey and publickey in the log which may not be desirable.  If we really need to make this method public, we can add @JsonIgnore annotation to this method.

Reply:
I left this method public because I wanted to extract the public key to sign payloads of a transaction to give it to constructionCombine of cardano-rosetta.
But you're right, it's better to keep it private to have a good implementation.